api/common/src/util.rs

```diff
@@ -0,0 +1,78 @@
+use bstr::BStr;
+use std::alloc::Layout;
+use std::marker::PhantomData;
+use std::slice;
+
+/// A string reference which may be passed over FFI boundaries.
+///
+/// Does not necessarily point to valid UTF8.
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct FStr<'a> {
+    ptr: *const u8,
+    len: usize,
+    _lifetime: PhantomData<&'a [u8]>,
```

Comment:
is `[u8]` prefered over the `()` unit type?

Reply:
The type of PhantomData is supposed to indicate what the phantom lifetime is used for—in this case, the lifetime is used for `ptr`, which is a pointer to a slice of `u8`.